Packs/Reco/Integrations/Reco/Reco.py

```diff
@@ -317,18 +318,74 @@ def resolve_visibility_event(self, entity_id: str, label_name: str) -> Any:
+        return self.get_identities(email_address=None, label=RISKY_USER)
+
+    def get_identities(self, email_address: Optional[str] = None, label: Optional[str] = None) -> Dict[str, Any]:
+        """
+        Get identities from Reco with specified filters.
+
+        :param email_address: Optional email substring to filter identities.
+        :param label: Optional label value to filter identities.
+        :return: A dictionary representing the getTableRequest payload.
+        """
         params = {
             "getTableRequest": {
-                "tableName": "RISK_MANAGEMENT_VIEW_USER_LIST",
-                "pageSize": 200,
+                "tableName": "RISK_MANAGEMENT_VIEW_IDENTITIES",
+                "pageSize": 50,
                 "fieldSorts": {
                     "sorts": [
-                        {"sortBy": "risk_level", "sortDirection": "SORT_DIRECTION_DESC"}
+                        {"sortBy": "primary_email_address", "sortDirection": "SORT_DIRECTION_ASC"}
                     ]
                 },
-                "fieldFilters": {},
+                "fieldFilters": {
+                    "relationship": "FILTER_RELATIONSHIP_AND",
+                    "fieldFilterGroups": {
+                        "fieldFilters": []
+                    },
+                    "forceEstimateSize": True
+                },
             }
         }
+
+        # Add label filter if provided
+        if label is not None:
+            label_filter = {
+                "relationship": "FILTER_RELATIONSHIP_OR",
+                "filters": {
+                    "filters": [
+                        {
+                            "field": "labels",
+                            "labelNameEquals": {
+                                "keys": ["identity_id"],
+                                "value": [label],
+                                "filterColumn": "label_name",
+                                "entryTypes": ["ENTRY_TYPE_IDENTITY"]
+                            }
+                        }
+                    ]
+                }
+            }
+            params["getTableRequest"]["fieldFilters"]["fieldFilterGroups"]["fieldFilters"].append(label_filter)
+
+        # Add email address filter if provided
+        if email_address is not None:
+            email_filter = {
+                "relationship": "FILTER_RELATIONSHIP_OR",
+                "filters": {
+                    "filters": [
+                        {
+                            "field": "full_name",
+                            "stringContains": {"value": email_address}
+                        },
+                        {
+                            "field": "primary_email_address",
+                            "stringContains": {"value": email_address}
+                        }
+                    ]
+                }
+            }
+            params["getTableRequest"]["fieldFilters"]["fieldFilterGroups"]["fieldFilters"].append(email_filter)
```

Comment:
If we add a handler function like this then it may make the label creation step a bit easier to read and extend in the future.

```
def create_filter(field, value):
        return {"field": field, "stringContains": {"value": value}}
```

Then it would be used like this
```suggestion
   if email_address:
           email_filter = {
               "relationship": "FILTER_RELATIONSHIP_OR",
               "filters": {
                   "filters": [
                       create_filter("full_name", email_address),
                       create_filter("primary_email_address", email_address)
                   ]
               }
           }
           params["getTableRequest"]["fieldFilters"]["fieldFilterGroups"]["fieldFilters"].append(email_filter)
```

This isn't a must, but worth looking into if you agree.